config/Config-images.in

```diff
@@ -20,6 +20,7 @@ menu "Target Images"
 			default TARGET_INITRAMFS_COMPRESSION_LZMA if TARGET_lantiq
 			default TARGET_INITRAMFS_COMPRESSION_LZMA if TARGET_mpc85xx
 			default TARGET_INITRAMFS_COMPRESSION_LZMA if TARGET_ramips
+			default TARGET_INITRAMFS_COMPRESSION_LZMA if TARGET_ipq40xx
```

Comment:
Typically, enabling this is not really desirable, as it may have side effects for other devices in the target.

Reply:
Fixed.

Reply:
no we must make sure CONFIG_INITRAMFS_COMPRESSION_LZMA is being set in kernel config

Reply:
Is it a requirement from ASUS?

Reply:
@cybik  we need a initramfs kernel image

Reply:
That option only defines the compression as LZMA, I believe.

@adrianschmutzler is there a way to force `TARGET_INITRAMFS_COMPRESSION_LZMA` to `y` for specific devices through the Makefile?

Reply:
I'm no expert with this stuff, but that's how I understand it:
There is two layers/levels of LZMA compression, one being the mentioned here, and the other being achieved by appropriate Build repices in image/Makefile. If you activate both, compression will be done _twice_; this will obviously not have a substantial additional size reduction effect, but it will cause the image to be decompressed in two steps. This might be a problem for some devices which have lzma enabled already via Build recipes.
Therefore, if you really need to enable this target-wide, you will have to take care that none of the other devices will break.

Reply:
by default no devices use initramfs-kernel-image and it affects only initramfs-kernel-image not the kernel-image
and TARGET_INITRAMFS_COMPRESSION_LZMA could be enable by user in `make menuconfig` if needed
here a initramfs-kernel-image is needed to build a  factory.trx for AC58U so we have to enable USE_INITRAMFS
and set TARGET_INITRAMFS_COMPRESSION_LZMA to make sure initramfs-kernel-image is lzma compressed to reduce the size.

Reply:
@adrianschmutzler that's exactly why I'm asking whether we can force TARGET_INITRAMFS_COMPRESSION_LZMA (and USE_INITRAMFS) through the AC58U configuration block.